apps/gitness/src/pages/logout.tsx

```diff
@@ -1,16 +1,25 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import { useNavigate } from 'react-router-dom'
-import useToken from '../framework/hooks/useToken'
 import { Button } from '@harnessio/canary'
+import { useOpLogoutMutation } from '@harnessio/code-service-client'
+import { useAppContext } from '../framework/context/AppContext'
 
 export const Logout: React.FC = () => {
+  const { resetApp } = useAppContext()
   const navigate = useNavigate()
-  const { removeToken } = useToken()
+  const { mutate: logout, isSuccess } = useOpLogoutMutation({})
+
+  useEffect(() => {
+    if (isSuccess) {
+      resetApp()
+      navigate('/signin') // Redirect to sign-in page
+    }
+  }, [isSuccess])
 
   const handleLogout = () => {
-    removeToken() // Remove the token when the user logs out
-    navigate('/signin') // Redirect to sign-in page
+    logout({})
```

Comment:
should we add error handling on this ?

Reply:
Yes we would, when we eventually has `toaster` integrated.